manager/constraint/constraint.go

```diff
@@ -125,6 +125,26 @@ func NodeMatches(constraints []Constraint, n *api.Node) bool {
 			if !constraint.Match(n.Spec.Role.String()) {
 				return false
 			}
+		case strings.EqualFold(constraint.key, "node.platform.os"):
+			if n.Description == nil || n.Description.Platform == nil {
+				if !constraint.Match("") {
+					return false
+				}
```

Comment:
I'd think that if you try to impose an `os` or `arch` constraint on a node that hasn't supplied a description it should always fail, but I don't have a strong feeling about this.